### PR TITLE
Add GitHub Actions workflow for building the project on Windows

### PR DESCRIPTION
**Pull request by xoascf**
This PR adds a workflow for building the project on Windows.
It includes steps to setup OpenSSL, build the project using Visual Studio, and copy the required DLLs. This automates the build process on Windows and improves the development process.
~I used a different host for the OpenSSL setup file because it took a long time to download from the Internet Archive (but once it's cached, it shouldn't be a problem!).~ Using https://github.com/DiscordMessenger/openssl now!

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        id: check-cache
+        uses: actions/cache@v4
+        with:
+          path: Win32OpenSSL-3_0_1.exe
+          key: '${{ runner.os }}-openssl-installer'
+
+      - name: Download and extract OpenSSL
+        if: steps.check-cache.outputs.cache-hit != 'true'
+        run: |
+          curl -o OpenSSL-Win32.zip "https://files.catbox.moe/xtfk7t.zip"
```
Comment (iProgramMC) [inline .github/workflows/build.yml:26]: When I put out a release for OpenSSL for MSVC (at https://github.com/DiscordMessenger/openssl), please use that instead.

**Review by iProgramMC — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (xoascf) [reply .github/workflows/build.yml:26]: .lib files are missing from the release.

**Comment by iProgramMC** [comment]
Ok, I've pushed the MSVC version of OpenSSL as a release on the linked repo. You should now use it.

File: .github/workflows/build.yml (hunk 1 above)
Comment (iProgramMC) [reply .github/workflows/build.yml:26]: > ~.lib files are missing from the release.~

Yes, those were intended for MinGW only. I just added the MSVC library files.

**Review by iProgramMC — APPROVED** [review]

**Review by iProgramMC — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -46,15 +22,10 @@ jobs:
...
-          path: Win32OpenSSL-3_0_1.exe
-          key: '${{ runner.os }}-openssl-installer'
-
-      - name: Install OpenSSL
-        if: steps.cache-openssl.outputs.cache-hit != 'true'
-        shell: cmd
-        run: start /wait Win32OpenSSL-3_0_1.exe /VERYSILENT /DIR="${{ env.OPENSSL_INSTALL }}"
+      - name: Setup OpenSSL
+        run: |
+          curl -Lo "openssl-prebuilt-msvc.zip" https://github.com/DiscordMessenger/openssl/releases/download/release-1-msvc/openssl-prebuilt-msvc.zip
```
[the remark this answers is not in the document]
Comment (xoascf) [reply .github/workflows/build.yml:27]: It's not tagged as latest, so I had to use the `release-1-msvc` release tag.

[the remark this answers is not in the document]
Comment (xoascf) [reply .github/workflows/build.yml:27]: Will try anyway to find a way...

[the remark this answers is not in the document]
Comment (iProgramMC) [reply .github/workflows/build.yml:27]: Ok, makes sense.

**Review by iProgramMC — APPROVED** [review]

**Pull request merged**